projects/super_res/super_res.py

```diff
@@ -0,0 +1,203 @@
+import xarray as xr
+import numpy as np
+import fv3viz
+from vcm.catalog import catalog
+import matplotlib.pyplot as plt
+import os
+import cartopy.crs as ccrs
+
+GRID = catalog["grid/c384"].read()
+
+DPI = 800
+
+
+def repeat_arr(arr, upscale_factor):
+    # This repeats values along the last dimension
+    return np.repeat(arr, upscale_factor, axis=-1)
+
+
+def upscale_nearest(ds, upscale_factor, dim):
+    ds_upscaled = xr.apply_ufunc(
+        repeat_arr,
+        ds,
+        input_core_dims=[[dim]],  # apply function along this dimension
+        output_core_dims=[[dim]],  # the output will also have this dimension
+        kwargs={
+            "upscale_factor": upscale_factor
+        },  # pass in other arguments to the function
+        dask="parallelized",  # needed for dask arrays
+        output_sizes={
+            dim: ds.sizes[dim] * upscale_factor
+        },  # the size of the output dimension
+        exclude_dims=set((dim,)),  # allow changing the dimension size
+    )
+    return ds_upscaled
+
+
+def plot_pattern_bias(output: xr.Dataset, target: xr.Dataset, varname: str):
+    """Plot the pattern bias between two datasets
+
+    Args:
+        output (xr.Dataset): output dataset
+        target (xr.Dataset): target dataset
+    """
+    # compute the pattern bias
+    out_mean: xr.Dataset = output.mean("time")
+    target_mean: xr.Dataset = target.mean("time")
+    vmin = min(
```

Comment:
If you make computing the pattern bias a separate function, I think that would helpful clarify this a lot. e.g. it looks like vmin and vmax are only used for plotting.

Reply:
I moved the `bias` calculation up to make this more explicit. I generally prefer having long plotting functions, but I'm going to leave it up to @cynicalsavant how he wants to build/modify this script and the style he wants to set for it.

Reply:
Sounds good, although a separate function might make it more elegant. In any case, I'll adapt it to make it more suitable to my needs in the coming future. Nevertheless, it's super helpful as a starting point.

Reply:
Yeah, whatever you find easiest! Glad to help :)